pkg/proto/eth_transaction.go

```diff
@@ -458,6 +477,98 @@ func (tx *EthereumTransaction) RawSignatureValues() (v, r, s *big.Int) {
 	return tx.inner.rawSignatureValues()
 }
 
+func validateEthereumTx(tx *EthereumTransaction) error {
+	switch tx.TxKind.(type) {
+	case *EthereumTransferWavesTxKind:
+		res := new(big.Int).Div(tx.inner.value(), big.NewInt(int64(DiffEthWaves)))
+		if ok := res.IsInt64(); !ok {
+			return errors.Errorf("failed to convert amount from ethreum transaction (big int) to int64. value is %d", tx.Value())
+		}
```

Comment:
You can easily use previous version for of `EthereumWeiToWavelet` function for conversion.